decidim-comments/app/queries/decidim/comments/metrics/comments_metric_manage.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Comments
+    module Metrics
+      class CommentsMetricManage < Decidim::MetricManage
+        def metric_name
+          "comments"
+        end
+
+        def save
+          return @registry if @registry
+
+          @registry = []
+          query.each do |key, results|
+            cumulative_value = results[:cumulative]
+            next if cumulative_value.zero?
+            quantity_value = results[:quantity] || 0
+            space_type, space_id, category_id, related_object_type, related_object_id = key
+            record = Decidim::Metric.find_or_initialize_by(day: @day.to_s, metric_type: @metric_name,
+                                                           participatory_space_type: space_type, participatory_space_id: space_id,
+                                                           organization: @organization, decidim_category_id: category_id,
+                                                           related_object_type: related_object_type, related_object_id: related_object_id)
+            record.assign_attributes(cumulative: cumulative_value, quantity: quantity_value)
+            @registry << record
+          end
+          @registry.each(&:save!)
+          @registry
+        end
+
+        private
+
+        def query
+          return @query if @query
+
+          user_ids = Decidim::User.select(:id).where(organization: @organization).collect(&:id)
+          user_group_ids = Decidim::UserGroup.select(:id).where(organization: @organization).collect(&:id)
+          comments = Decidim::Comments::Comment.includes(:root_commentable).not_hidden
+                                               .where("decidim_comments_comments.created_at <= ?", end_time)
+                                               .where("decidim_comments_comments.decidim_author_id IN (?) OR
+                                                     decidim_comments_comments.decidim_user_group_id IN (?)", user_ids, user_group_ids)
+
+          @query = comments.each_with_object({}) do |comment, query_h|
```

Comment:
Please find a better name for `query_h`, I don't know what's supposed to represent.

Reply:
Ok